### PR TITLE
Ncsd 2066 deploy aks

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: PSScripts/New-AksServicePrincipals.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+}
+else {
+
+    Write-Verbose "Assigning 'Contributor' to $($AksServicePrincipal.Id)"
+    New-AzureRmRoleAssignment -ObjectId $AksServicePrincipal.Id -RoleDefinitionName Contributor -Scope "/subscriptions/$($Context.Subscription.Id)"
+
+}
+
+# Create Service Principl with Delegated Permissions Directory.Read.All & User.Read and Application Permissions Directory.Read.All
```
Comment (thelious1974) [inline PSScripts/New-AksServicePrincipals.ps1]: typo on `Principl`

File: Resources/AzureDevOps/azure-pipelines.yml [hunk 2]
```diff
@@ -14,20 +12,66 @@ pr:
...
+      displayName: Publish Build Artifact
+      inputs:
+        pathtoPublish: $(Build.ArtifactStagingDirectory)
+        artifactName: drop
+
+- stage: DeployToDev
+  variables:
+  - group: dfc-shared-all
+  - group: dfc-shared-infrastructure-all
+  - template: VariableTemplates\ReleaseVariables.yml
```
Comment (thelious1974) [inline Resources/AzureDevOps/azure-pipelines.yml]: I don't think this should be here!

File: PSScripts/New-AksServicePrincipals.ps1 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    [Parameter(Mandatory=$true)]
+    [String]$AksServicePrincipalName,
+    [Parameter(Mandatory=$true)]
+    [String]$AksAdClientApplicationName,
+    [Parameter(Mandatory=$true)]
+    [String]$AksAdServerApplicationName,
+    [Parameter(Mandatory=$true)]
+    [String]$DfcDevOpsScriptRoot,
+    [Parameter(Mandatory=$true)]
+    [String]$SharedKeyVaultName
```
Comment (thelious1974) [inline PSScripts/New-AksServicePrincipals.ps1]: doc string for this parameter are missing

File: Resources/AzureDevOps/azure-pipelines.yml (hunk 2 above)
Comment (thelious1974) [reply Resources/AzureDevOps/azure-pipelines.yml]: Ignore that, the variable templates have different variables in them.
Maybe we need to look at their filename as not to cause confusion?

File: PSScripts/New-AksServicePrincipals.ps1 (hunk 3 above)
Comment (NickGraham101) [reply PSScripts/New-AksServicePrincipals.ps1]: Added documentation

File: PSScripts/New-AksServicePrincipals.ps1 (hunk 1 above)
Comment (NickGraham101) [reply PSScripts/New-AksServicePrincipals.ps1]: Fixed type

File: Resources/AzureDevOps/azure-pipelines.yml (hunk 2 above)
Comment (NickGraham101) [reply Resources/AzureDevOps/azure-pipelines.yml]: Renamed files

**Comment by NickGraham101** [comment]
This PR is being left open as additional preparatory work is required to deploy AKS to PP.  Until that is completed these changes should not be merged to master

File: Resources/AzureDevOps/JobTemplates/Deploy.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  ResourceGroup: ''
+  ParentBusinessTag: ''
+  ServiceOfferingTag: ''
+  SharedKeyVaultName: ''
+  SharedPrefix: ''
+  StorageAccountName: ''
+  
+jobs:
+- deployment: DeployTo_${{ parameters.Environment }}
+  pool:
```
Comment (thelious1974) [inline Resources/AzureDevOps/JobTemplates/Deploy.yml]: Suggestion: I'd be tempted to pull this out to the top level, so it affects everything

File: Resources/AzureDevOps/azure-pipelines.yml [hunk 5]
```diff
@@ -7,27 +9,113 @@ resources:
...
+    - task: TotalALM.totalalm-tokenization.tokenization-build-task.Tokenization@2
+      displayName: 'Tokenization: Transform file parameters.json'
+      inputs:
+        SourcePath: '$(System.DefaultWorkingDirectory)\Resources'
+        TargetFileNames: test-parameters.json
+    - template: AzureDevOpsTemplates/Build/StepTemplates/dfc-arm-build.yml@dfc-devops
+      parameters:
+        ArmTemplateRoot: '$(System.DefaultWorkingDirectory)\Resources'
+        SolutionBaseName: $(SolutionBaseName)
+        PublishPipelineArtifact: true
```
Comment (thelious1974) [inline Resources/AzureDevOps/azure-pipelines.yml]: Suggestion:  If you bump the dfc-devops version to v1.10.1,  you no longer need to specify the PublishPipelineArtifact parameter here.

File: Resources/AzureDevOps/azure-pipelines.yml [hunk 6]
```diff
@@ -7,27 +9,113 @@ resources:
...
+      inputs:
+        Contents: 'PSScripts/**/*.ps1'
+        TargetFolder: '$(Build.ArtifactStagingDirectory)/Resources/PSScripts'
+        flattenFolders: true
+    - task: PublishPipelineArtifact@0
+      displayName: Publish Pipeline Artifact
+      inputs:
+        targetPath: $(Build.ArtifactStagingDirectory)/Resources/PSScripts
+        artifactName: Dfc.Shared.Resources.PSScripts
+    # Also publish build artifact for backwards compatibility with classic Release Definition
```
Comment (thelious1974) [inline Resources/AzureDevOps/azure-pipelines.yml]: FYI:  "classic" release definitions can see and use pipeline artifacts, so this really isn't needed

File: Resources/AzureDevOps/JobTemplates/Deploy.yml (hunk 4 above)
Comment (NickGraham101) [reply Resources/AzureDevOps/JobTemplates/Deploy.yml]: Moved it

File: Resources/AzureDevOps/azure-pipelines.yml (hunk 5 above)
Comment (NickGraham101) [reply Resources/AzureDevOps/azure-pipelines.yml]: bumped and removed unneeded param

File: Resources/AzureDevOps/azure-pipelines.yml (hunk 6 above)
Comment (NickGraham101) [reply Resources/AzureDevOps/azure-pipelines.yml]: Agreed, I will leave this comment open as a reminder

File: README.md [hunk 7]
```diff
@@ -8,11 +8,32 @@ Each environment (DEV, LAB, SIT, PP and PRD) will have its own shared resources.
 Deployed using the comnbination of a template file and a parameters file.
 
 Two parameters files must be maintained:
-    * [parameters.json](parameters.json) is used in the Azure DevOps pipeline and has tokenized values managed within Azure DevOps.
-    * [test-parameters.json](test-parameters.json) for local and test deployment.
+    * [parameters.json](Resources/parameters.json) is used in the Azure DevOps pipeline and has tokenized values managed within Azure DevOps.
+    * [test-parameters.json](Resources/template.json) for local and test deployment.
```
Comment (rivoric) [inline README.md]: Should this say [parameters.json](Resources/template.json)

**Pull request merged**